{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "022fb35d",
   "metadata": {},
   "source": [
    "# Preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "0db3f0cb",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:46:36.920240Z",
     "start_time": "2021-08-25T07:46:36.912055Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import sklearn\n",
    "from scipy import stats\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.preprocessing import MinMaxScaler, StandardScaler\n",
    "from sklearn.pipeline import Pipeline, make_pipeline\n",
    "from sklearn.compose import ColumnTransformer, make_column_transformer, make_column_selector\n",
    "from sklearn.impute import SimpleImputer, KNNImputer\n",
    "from sklearn.preprocessing import RobustScaler, OneHotEncoder, OrdinalEncoder\n",
    "from sklearn.metrics import make_scorer\n",
    "from sklearn.model_selection import cross_val_score\n",
    "from sklearn.neighbors import KNeighborsRegressor\n",
    "from sklearn.linear_model import Ridge, Lasso, LinearRegression\n",
    "from sklearn.model_selection import RandomizedSearchCV, cross_validate\n",
    "from sklearn.svm import SVR\n",
    "from sklearn.model_selection import GridSearchCV, RandomizedSearchCV\n",
    "from sklearn.ensemble import AdaBoostRegressor\n",
    "from sklearn.tree import DecisionTreeRegressor\n",
    "from sklearn.ensemble import VotingRegressor\n",
    "from sklearn.ensemble import GradientBoostingRegressor\n",
    "from sklearn.ensemble import StackingRegressor\n",
    "from sklearn.ensemble import RandomForestRegressor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "c03ad356",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:41:13.863203Z",
     "start_time": "2021-08-25T07:41:13.820525Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>jour</th>\n",
       "      <th>service</th>\n",
       "      <th>CA_TTC</th>\n",
       "      <th>CA_HT</th>\n",
       "      <th>TVA</th>\n",
       "      <th>temp</th>\n",
       "      <th>feels_like</th>\n",
       "      <th>temp_min</th>\n",
       "      <th>temp_max</th>\n",
       "      <th>wind_speed</th>\n",
       "      <th>clouds_all</th>\n",
       "      <th>weather_main</th>\n",
       "      <th>weather_description</th>\n",
       "      <th>match_edf</th>\n",
       "      <th>roland_garros</th>\n",
       "      <th>fashion_week</th>\n",
       "      <th>Match Happening</th>\n",
       "      <th>Match Happening-CL</th>\n",
       "      <th>vacances_paris</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>Dimanche</td>\n",
       "      <td>midi</td>\n",
       "      <td>3548.00</td>\n",
       "      <td>3191.82</td>\n",
       "      <td>356.18</td>\n",
       "      <td>294.08</td>\n",
       "      <td>293.35</td>\n",
       "      <td>292.94</td>\n",
       "      <td>295.87</td>\n",
       "      <td>2.68</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Clear</td>\n",
       "      <td>sky is clear</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>Dimanche</td>\n",
       "      <td>soir</td>\n",
       "      <td>4577.10</td>\n",
       "      <td>4035.30</td>\n",
       "      <td>541.80</td>\n",
       "      <td>293.37</td>\n",
       "      <td>292.60</td>\n",
       "      <td>291.95</td>\n",
       "      <td>294.40</td>\n",
       "      <td>4.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Clear</td>\n",
       "      <td>sky is clear</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2019-09-02</td>\n",
       "      <td>Lundi</td>\n",
       "      <td>midi</td>\n",
       "      <td>2089.00</td>\n",
       "      <td>1888.79</td>\n",
       "      <td>200.21</td>\n",
       "      <td>294.26</td>\n",
       "      <td>293.47</td>\n",
       "      <td>292.71</td>\n",
       "      <td>295.87</td>\n",
       "      <td>2.60</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Clear</td>\n",
       "      <td>sky is clear</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2019-09-02</td>\n",
       "      <td>Lundi</td>\n",
       "      <td>soir</td>\n",
       "      <td>5726.50</td>\n",
       "      <td>5056.44</td>\n",
       "      <td>670.06</td>\n",
       "      <td>294.00</td>\n",
       "      <td>293.08</td>\n",
       "      <td>292.25</td>\n",
       "      <td>295.36</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Clear</td>\n",
       "      <td>sky is clear</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2019-09-03</td>\n",
       "      <td>Mardi</td>\n",
       "      <td>midi</td>\n",
       "      <td>1745.25</td>\n",
       "      <td>1567.27</td>\n",
       "      <td>177.98</td>\n",
       "      <td>295.07</td>\n",
       "      <td>294.57</td>\n",
       "      <td>293.18</td>\n",
       "      <td>296.42</td>\n",
       "      <td>0.45</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Clouds</td>\n",
       "      <td>scattered clouds</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         date      jour service   CA_TTC    CA_HT     TVA    temp  feels_like  \\\n",
       "0  2019-09-01  Dimanche    midi  3548.00  3191.82  356.18  294.08      293.35   \n",
       "1  2019-09-01  Dimanche    soir  4577.10  4035.30  541.80  293.37      292.60   \n",
       "2  2019-09-02     Lundi    midi  2089.00  1888.79  200.21  294.26      293.47   \n",
       "3  2019-09-02     Lundi    soir  5726.50  5056.44  670.06  294.00      293.08   \n",
       "4  2019-09-03     Mardi    midi  1745.25  1567.27  177.98  295.07      294.57   \n",
       "\n",
       "   temp_min  temp_max  wind_speed  clouds_all weather_main  \\\n",
       "0    292.94    295.87        2.68         0.0        Clear   \n",
       "1    291.95    294.40        4.02         0.0        Clear   \n",
       "2    292.71    295.87        2.60         0.0        Clear   \n",
       "3    292.25    295.36        0.45         0.0        Clear   \n",
       "4    293.18    296.42        0.45        40.0       Clouds   \n",
       "\n",
       "  weather_description  match_edf  roland_garros  fashion_week  \\\n",
       "0        sky is clear        0.0            0.0           0.0   \n",
       "1        sky is clear        0.0            0.0           0.0   \n",
       "2        sky is clear        0.0            0.0           0.0   \n",
       "3        sky is clear        0.0            0.0           0.0   \n",
       "4    scattered clouds        0.0            0.0           0.0   \n",
       "\n",
       "   Match Happening  Match Happening-CL  vacances_paris  \n",
       "0              NaN                 NaN             1.0  \n",
       "1              NaN                 NaN             NaN  \n",
       "2              0.0                 NaN             0.0  \n",
       "3              NaN                 NaN             NaN  \n",
       "4              NaN                 NaN             0.0  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset_d16 = '/Users/monmac/Desktop/preproc_data_d16.csv'\n",
    "dataset_d2 = '/Users/monmac/Desktop/preproc_data_d2.csv'\n",
    "data_d16 = pd.read_csv(dataset_d16)\n",
    "data_d2 = pd.read_csv(dataset_d2)\n",
    "data_d16.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "07f71f16",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:41:14.991323Z",
     "start_time": "2021-08-25T07:41:14.980794Z"
    }
   },
   "outputs": [],
   "source": [
    "data_d16.fillna(value = \"0.0\", inplace = True)\n",
    "data_d2.fillna(value = \"0.0\", inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b91b51f0",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:28:32.925516Z",
     "start_time": "2021-08-25T07:28:32.915699Z"
    }
   },
   "outputs": [],
   "source": [
    "weather_description = pd.DataFrame(np.concatenate([data_d16.weather_description.unique(),data_d2.weather_description.unique()])).drop_duplicates()\n",
    "weather_description.sort_values(by=0,axis=0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a95ef0d6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:28:49.787665Z",
     "start_time": "2021-08-25T07:28:49.763076Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clear</th>\n",
       "      <th>Clouds</th>\n",
       "      <th>Drizzle</th>\n",
       "      <th>Drizzle and Rain</th>\n",
       "      <th>Fog</th>\n",
       "      <th>Haze</th>\n",
       "      <th>Mist</th>\n",
       "      <th>Rain</th>\n",
       "      <th>Snow</th>\n",
       "      <th>Thunderstorm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sky is clear</td>\n",
       "      <td>scattered clouds</td>\n",
       "      <td>light intensity drizzle</td>\n",
       "      <td>light intensity drizzle rain</td>\n",
       "      <td>fog</td>\n",
       "      <td>haze</td>\n",
       "      <td>mist</td>\n",
       "      <td>light rain</td>\n",
       "      <td>light snow</td>\n",
       "      <td>proximity thunderstorm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>few clouds</td>\n",
       "      <td>drizzle</td>\n",
       "      <td>rain and drizzle</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>light intensity shower rain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>thunderstorm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>broken clouds</td>\n",
       "      <td>heavy intensity drizzle</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moderate rain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>thunderstorm with light rain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>overcast clouds</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>heavy intensity rain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>thunderstorm with heavy rain</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Clear            Clouds                  Drizzle  \\\n",
       "0  sky is clear  scattered clouds  light intensity drizzle   \n",
       "1           NaN        few clouds                  drizzle   \n",
       "2           NaN     broken clouds  heavy intensity drizzle   \n",
       "3           NaN   overcast clouds                      NaN   \n",
       "\n",
       "               Drizzle and Rain  Fog  Haze  Mist                         Rain  \\\n",
       "0  light intensity drizzle rain  fog  haze  mist                   light rain   \n",
       "1              rain and drizzle  NaN   NaN   NaN  light intensity shower rain   \n",
       "2                           NaN  NaN   NaN   NaN                moderate rain   \n",
       "3                           NaN  NaN   NaN   NaN         heavy intensity rain   \n",
       "\n",
       "         Snow                  Thunderstorm  \n",
       "0  light snow        proximity thunderstorm  \n",
       "1         NaN                  thunderstorm  \n",
       "2         NaN  thunderstorm with light rain  \n",
       "3         NaN  thunderstorm with heavy rain  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "weather_sorted = {\n",
    "'Clear' : ['sky is clear', np.nan, np.nan,np.nan], \n",
    "'Clouds' : ['scattered clouds','few clouds','broken clouds','overcast clouds'],\n",
    "'Drizzle' : ['light intensity drizzle','drizzle','heavy intensity drizzle',np.nan], \n",
    "'Drizzle and Rain' : ['light intensity drizzle rain','rain and drizzle',np.nan,np.nan],\n",
    "'Fog' : ['fog', np.nan, np.nan,np.nan], \n",
    "'Haze' : ['haze', np.nan, np.nan,np.nan], \n",
    "'Mist' : ['mist', np.nan, np.nan,np.nan], \n",
    "'Rain' : ['light rain','light intensity shower rain','moderate rain', 'heavy intensity rain'], \n",
    "'Snow' : ['light snow', np.nan, np.nan,np.nan], \n",
    "'Thunderstorm' : ['proximity thunderstorm', 'thunderstorm', 'thunderstorm with light rain','thunderstorm with heavy rain'] \n",
    "}\n",
    "\n",
    "weather_sorted_df = pd.DataFrame(data=weather_sorted)\n",
    "weather_sorted_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1ff7b742",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:35:23.914569Z",
     "start_time": "2021-08-25T07:35:23.892116Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CA_TTC</th>\n",
       "      <td>1297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CA_HT</th>\n",
       "      <td>1314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TVA</th>\n",
       "      <td>1308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>temp</th>\n",
       "      <td>1040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>feels_like</th>\n",
       "      <td>1081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>temp_min</th>\n",
       "      <td>605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>temp_max</th>\n",
       "      <td>676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wind_speed</th>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>clouds_all</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>match_edf</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>roland_garros</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fashion_week</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               unique_values\n",
       "CA_TTC                  1297\n",
       "CA_HT                   1314\n",
       "TVA                     1308\n",
       "temp                    1040\n",
       "feels_like              1081\n",
       "temp_min                 605\n",
       "temp_max                 676\n",
       "wind_speed                43\n",
       "clouds_all                 7\n",
       "match_edf                  2\n",
       "roland_garros              2\n",
       "fashion_week               2"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feat_numerical_nunique_d2 = pd.DataFrame(data_d2.select_dtypes(exclude=['object'],include=['int64','float64']).nunique(), columns = [\"unique_values\"])\n",
    "feat_numerical_nunique_d2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "77c1deb8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:35:29.955730Z",
     "start_time": "2021-08-25T07:35:29.939106Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_values</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>date</th>\n",
       "      <td>672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>jour</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>service</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>weather_main</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>weather_description</th>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Match Happening</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Match Happening-CL</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vacances_paris</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     unique_values\n",
       "date                           672\n",
       "jour                             7\n",
       "service                          2\n",
       "weather_main                     8\n",
       "weather_description             21\n",
       "Match Happening                  2\n",
       "Match Happening-CL               2\n",
       "vacances_paris                   3"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feat_categorical_nunique_d2 = pd.DataFrame(data_d2.select_dtypes(include=['object'],exclude=['int64','float64']).nunique(), columns = [\"unique_values\"])\n",
    "feat_categorical_nunique_d2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b53f9ad",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "65022c11",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:41:09.414653Z",
     "start_time": "2021-08-25T07:41:09.411126Z"
    }
   },
   "outputs": [],
   "source": [
    "#data.select_dtypes(include=['object'],exclude=['int64','float64']).nunique().index.sort_values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "56f968f0",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:43:58.714876Z",
     "start_time": "2021-08-25T07:43:58.705379Z"
    }
   },
   "outputs": [],
   "source": [
    "data_d16['Match Happening'] = pd.to_numeric(data_d16['Match Happening'])\n",
    "data_d16['Match Happening-CL'] = pd.to_numeric(data_d16['Match Happening-CL'])\n",
    "data_d16['vacances_paris'] = pd.to_numeric(data_d16['vacances_paris'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "28f29ed4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:44:19.261376Z",
     "start_time": "2021-08-25T07:44:19.248259Z"
    }
   },
   "outputs": [],
   "source": [
    "data_d2['Match Happening'] = pd.to_numeric(data_d2['Match Happening'])\n",
    "data_d2['Match Happening-CL'] = pd.to_numeric(data_d2['Match Happening-CL'])\n",
    "data_d2['vacances_paris'] = pd.to_numeric(data_d2['vacances_paris'])\n",
    "\n",
    "\n",
    "data_d2['vacances_paris'] = data_d2['vacances_paris'].astype(object)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "38c95c31",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T10:49:20.202449Z",
     "start_time": "2021-08-25T10:49:20.178384Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data_d2['vacances_paris'] = data_d2['vacances_paris'].astype(object)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "0c10ba98",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T10:49:26.956712Z",
     "start_time": "2021-08-25T10:49:26.946743Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "date                    object\n",
       "jour                    object\n",
       "service                 object\n",
       "CA_TTC                 float64\n",
       "CA_HT                  float64\n",
       "TVA                    float64\n",
       "temp                   float64\n",
       "feels_like             float64\n",
       "temp_min               float64\n",
       "temp_max               float64\n",
       "wind_speed             float64\n",
       "clouds_all             float64\n",
       "weather_main            object\n",
       "weather_description     object\n",
       "match_edf              float64\n",
       "roland_garros          float64\n",
       "fashion_week           float64\n",
       "Match Happening        float64\n",
       "Match Happening-CL     float64\n",
       "vacances_paris          object\n",
       "dtype: object"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_d2.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "416ec4ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "numerical_columns = \n",
    "categorical_columns = \n",
    "ordinal_columns ="
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "63e28543",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T08:44:19.188414Z",
     "start_time": "2021-08-25T08:44:19.132076Z"
    }
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'feat_categorical_small' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/p2/6ddwj2812811pbqjm8t6brv80000gn/T/ipykernel_95871/3770087919.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      9\u001b[0m preproc_baseline = make_column_transformer(\n\u001b[1;32m     10\u001b[0m     \u001b[0;34m(\u001b[0m\u001b[0mpreproc_numerical_baseline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmake_column_selector\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype_include\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"int64\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"float64\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 11\u001b[0;31m     \u001b[0;34m(\u001b[0m\u001b[0mpreproc_categorical_baseline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeat_categorical_small\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     12\u001b[0m     remainder=\"drop\")\n",
      "\u001b[0;31mNameError\u001b[0m: name 'feat_categorical_small' is not defined"
     ]
    }
   ],
   "source": [
    "preproc_numerical_baseline = make_pipeline(\n",
    "    RobustScaler())\n",
    "\n",
    "preproc_categorical_baseline = make_pipeline(\n",
    "    SimpleImputer(strategy=\"most_frequent\"),\n",
    "    OneHotEncoder(handle_unknown=\"ignore\"))\n",
    "\n",
    "preproc_baseline = make_column_transformer(\n",
    "    (preproc_numerical_baseline, make_column_selector(dtype_include=[\"int64\", \"float64\"])),\n",
    "    (preproc_categorical_baseline, make_column_selector(pattern=),\n",
    "    (preproc_ordinal_d16, make_column_selector(pattern=))\n",
    "    remainder=\"drop\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "f6f55cd6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T09:01:48.652048Z",
     "start_time": "2021-08-25T09:01:48.638521Z"
    }
   },
   "outputs": [],
   "source": [
    "d16 = pd.read_csv('/Users/monmac/Desktop/data_d16.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "01dbafa6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T09:40:46.396465Z",
     "start_time": "2021-08-25T09:40:46.361903Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "date                    object\n",
       "jour                    object\n",
       "service                 object\n",
       "CA_TTC                 float64\n",
       "CA_HT                  float64\n",
       "TVA                    float64\n",
       "temp                   float64\n",
       "feels_like             float64\n",
       "temp_min               float64\n",
       "temp_max               float64\n",
       "wind_speed             float64\n",
       "clouds_all             float64\n",
       "weather_main            object\n",
       "weather_description     object\n",
       "match_edf              float64\n",
       "roland_garros          float64\n",
       "fashion_week           float64\n",
       "Match Happening        float64\n",
       "Match Happening-CL     float64\n",
       "vacances_paris         float64\n",
       "Clear                   object\n",
       "Clouds                  object\n",
       "Drizzle                 object\n",
       "Drizzle and Rain        object\n",
       "Fog                     object\n",
       "Haze                    object\n",
       "Mist                    object\n",
       "Rain                    object\n",
       "Snow                     int64\n",
       "Thunderstorm            object\n",
       "dtype: object"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d16.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e708dc0d",
   "metadata": {},
   "source": [
    "# Model "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "00889fdc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T08:14:47.750431Z",
     "start_time": "2021-08-25T08:14:47.741458Z"
    }
   },
   "outputs": [],
   "source": [
    "X = data_d2.drop(labels=['CA_TTC', 'CA_HT','TVA'], axis=1)\n",
    "y = data_d2['CA_TTC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "72b97a98",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:52:55.950172Z",
     "start_time": "2021-08-25T07:52:55.942291Z"
    }
   },
   "outputs": [],
   "source": [
    "#plt.hist(data_d2['CA_TTC'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "0bb14657",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:49:18.139167Z",
     "start_time": "2021-08-25T07:49:18.043443Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-02-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/linear_model/_base.py\", line 518, in fit\n",
      "    X, y = self._validate_data(X, y, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "nan"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.linear_model import LinearRegression\n",
    "\n",
    "model = LinearRegression()\n",
    "\n",
    "cv_results = cross_validate(model, X, y, cv = 20)\n",
    "\n",
    "cv_score = cv_results['test_score'].mean()\n",
    "\n",
    "cv_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "67c4f122",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:52:58.364556Z",
     "start_time": "2021-08-25T07:52:58.358257Z"
    }
   },
   "outputs": [],
   "source": [
    "from sklearn.pipeline import Pipeline\n",
    "pipe = Pipeline([\n",
    "    ('imputer', SimpleImputer()),\n",
    "    ('scaler', RobustScaler()),\n",
    "    ('model', LinearRegression())\n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "d1a31ec5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:54:16.753178Z",
     "start_time": "2021-08-25T07:54:16.748767Z"
    }
   },
   "outputs": [],
   "source": [
    "from sklearn.model_selection import GridSearchCV\n",
    "grid_search = GridSearchCV(\n",
    "  pipe, \n",
    "  param_grid={'imputer__n_neighbors': [4,5,6]\n",
    "  },\n",
    "  cv=5,\n",
    "  scoring=\"accuracy\")\n",
    "\n",
    "#grid_search.fit(X, y)\n",
    "#grid_search.best_params_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "6dfe59cd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:54:42.955426Z",
     "start_time": "2021-08-25T07:54:42.939362Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'GridSearchCV' object has no attribute 'best_estimator_'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/p2/6ddwj2812811pbqjm8t6brv80000gn/T/ipykernel_95871/1808546380.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mtuned_pipe\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgrid_search\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbest_estimator_\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m: 'GridSearchCV' object has no attribute 'best_estimator_'"
     ]
    }
   ],
   "source": [
    "tuned_pipe = grid_search.best_estimator_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "524b54d2",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T07:54:25.792652Z",
     "start_time": "2021-08-25T07:54:25.777569Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'tuned_pipe' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/p2/6ddwj2812811pbqjm8t6brv80000gn/T/ipykernel_95871/63609103.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcv_results\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcross_validate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtuned_pipe\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcv\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m20\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mscoring\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"accuracy\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mbase_score\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcv_results\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'test_score'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmean\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mbase_score\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'tuned_pipe' is not defined"
     ]
    }
   ],
   "source": [
    "cv_results = cross_validate(tuned_pipe, X,y, cv = 20, scoring = \"accuracy\")\n",
    "\n",
    "base_score = cv_results['test_score'].mean()\n",
    "\n",
    "base_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "804d4840",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T08:14:39.054339Z",
     "start_time": "2021-08-25T08:14:39.031893Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "could not convert string to float: 'Mardi'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/p2/6ddwj2812811pbqjm8t6brv80000gn/T/ipykernel_95871/246425190.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0msklearn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtree\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mDecisionTreeClassifier\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mtree_clf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mDecisionTreeClassifier\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmax_depth\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrandom_state\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mtree_clf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, X, y, sample_weight, check_input, X_idx_sorted)\u001b[0m\n\u001b[1;32m    901\u001b[0m         \"\"\"\n\u001b[1;32m    902\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 903\u001b[0;31m         super().fit(\n\u001b[0m\u001b[1;32m    904\u001b[0m             \u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    905\u001b[0m             \u001b[0msample_weight\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msample_weight\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, X, y, sample_weight, check_input, X_idx_sorted)\u001b[0m\n\u001b[1;32m    155\u001b[0m             \u001b[0mcheck_X_params\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mDTYPE\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maccept_sparse\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"csc\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    156\u001b[0m             \u001b[0mcheck_y_params\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mensure_2d\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 157\u001b[0;31m             X, y = self._validate_data(X, y,\n\u001b[0m\u001b[1;32m    158\u001b[0m                                        validate_separately=(check_X_params,\n\u001b[1;32m    159\u001b[0m                                                             check_y_params))\n",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\u001b[0m in \u001b[0;36m_validate_data\u001b[0;34m(self, X, y, reset, validate_separately, **check_params)\u001b[0m\n\u001b[1;32m    428\u001b[0m                 \u001b[0;31m# :(\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    429\u001b[0m                 \u001b[0mcheck_X_params\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcheck_y_params\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvalidate_separately\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 430\u001b[0;31m                 \u001b[0mX\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcheck_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mcheck_X_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    431\u001b[0m                 \u001b[0my\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcheck_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mcheck_y_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    432\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36minner_f\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     61\u001b[0m             \u001b[0mextra_args\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mall_args\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mextra_args\u001b[0m \u001b[0;34m<=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 63\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     64\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m             \u001b[0;31m# extra_args > 0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36mcheck_array\u001b[0;34m(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator)\u001b[0m\n\u001b[1;32m    671\u001b[0m                     \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marray\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcasting\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"unsafe\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    672\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 673\u001b[0;31m                     \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0morder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    674\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mComplexWarning\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mcomplex_warning\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    675\u001b[0m                 raise ValueError(\"Complex data not supported\\n\"\n",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m__array__\u001b[0;34m(self, dtype)\u001b[0m\n\u001b[1;32m   1991\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1992\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__array__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mNpDtype\u001b[0m \u001b[0;34m|\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1993\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1994\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1995\u001b[0m     def __array_wrap__(\n",
      "\u001b[0;31mValueError\u001b[0m: could not convert string to float: 'Mardi'"
     ]
    }
   ],
   "source": [
    "from sklearn.tree import DecisionTreeClassifier\n",
    "tree_clf = DecisionTreeClassifier(max_depth=2, random_state=2)\n",
    "tree_clf.fit(X,y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "f568e69d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T08:20:44.582462Z",
     "start_time": "2021-08-25T08:20:44.532957Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-03-07'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 1252, in fit\n",
      "    super().fit(\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/tree/_classes.py\", line 157, in fit\n",
      "    X, y = self._validate_data(X, y,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 430, in _validate_data\n",
      "    X = check_array(X, **check_X_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    }
   ],
   "source": [
    "from sklearn.tree import DecisionTreeRegressor\n",
    "from sklearn.model_selection import cross_validate\n",
    "\n",
    "tree = DecisionTreeRegressor()\n",
    "\n",
    "cv_results = cross_validate(tree, X, y, scoring = \"r2\", cv=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "ea2df3ff",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T08:21:48.810745Z",
     "start_time": "2021-08-25T08:21:48.756365Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[nan nan nan nan nan nan nan nan nan nan]\n",
      "mean r2:  nan\n",
      "std r2:  nan\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-03-07'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n",
      "/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:615: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/ensemble/_forest.py\", line 304, in fit\n",
      "    X, y = self._validate_data(X, y, multi_output=True,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 871, in check_X_y\n",
      "    X = check_array(X, accept_sparse=accept_sparse,\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 673, in check_array\n",
      "    array = np.asarray(array, order=order, dtype=dtype)\n",
      "  File \"/Users/monmac/.pyenv/versions/3.8.6/envs/resto-project/lib/python3.8/site-packages/pandas/core/generic.py\", line 1993, in __array__\n",
      "    return np.asarray(self._values, dtype=dtype)\n",
      "ValueError: could not convert string to float: '2019-01-01'\n",
      "\n",
      "  warnings.warn(\"Estimator fit failed. The score on this train-test\"\n"
     ]
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestRegressor\n",
    "\n",
    "forest = RandomForestRegressor(n_estimators=100)\n",
    "\n",
    "cv_results = cross_validate(forest, X, y, scoring = \"r2\", cv=10)\n",
    "\n",
    "print(cv_results['test_score'])\n",
    "print('mean r2: ',cv_results['test_score'].mean())\n",
    "print('std r2: ', cv_results['test_score'].std())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "2e4f1f65",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T08:22:22.857470Z",
     "start_time": "2021-08-25T08:22:22.849420Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (357369733.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"/var/folders/p2/6ddwj2812811pbqjm8t6brv80000gn/T/ipykernel_95871/357369733.py\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    from xgboost import XGBRegressor()\u001b[0m\n\u001b[0m                                    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "from xgboost import XGBRegressor()\n",
    "xgb_reg.fit(\n",
    "    X_train, y_train,\n",
    "    # evaluate loss at each iteration\n",
    "    eval_set=[(X_val, y_val)], \n",
    "    # stop iterating when eval loss increases 5 times in a row\n",
    "    early_stopping_rounds=5\n",
    ") \n",
    "y_pred = xgb_reg.predict(X_val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "fe88b41b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T12:15:09.750697Z",
     "start_time": "2021-08-25T12:15:09.740939Z"
    }
   },
   "outputs": [],
   "source": [
    "d16=pd.read_csv('/Users/monmac/Desktop/preproc_data_d16.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "2cebf327",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-08-25T12:16:47.445350Z",
     "start_time": "2021-08-25T12:16:47.369210Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>date</th>\n",
       "      <th>jour</th>\n",
       "      <th>service</th>\n",
       "      <th>CA_TTC</th>\n",
       "      <th>CA_HT</th>\n",
       "      <th>TVA</th>\n",
       "      <th>temp</th>\n",
       "      <th>feels_like</th>\n",
       "      <th>temp_min</th>\n",
       "      <th>temp_max</th>\n",
       "      <th>wind_speed</th>\n",
       "      <th>clouds_all</th>\n",
       "      <th>weather_main</th>\n",
       "      <th>weather_description</th>\n",
       "      <th>match_edf</th>\n",
       "      <th>roland_garros</th>\n",
       "      <th>fashion_week</th>\n",
       "      <th>Match Happening</th>\n",
       "      <th>Match Happening-CL</th>\n",
       "      <th>vacances_paris</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>2019-09-18</td>\n",
       "      <td>Mercredi</td>\n",
       "      <td>midi</td>\n",
       "      <td>4077.50</td>\n",
       "      <td>3663.41</td>\n",
       "      <td>414.09</td>\n",
       "      <td>291.23</td>\n",
       "      <td>290.22</td>\n",
       "      <td>290.36</td>\n",
       "      <td>292.47</td>\n",
       "      <td>2.24</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Clear</td>\n",
       "      <td>sky is clear</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>2019-10-22</td>\n",
       "      <td>Mardi</td>\n",
       "      <td>midi</td>\n",
       "      <td>3671.75</td>\n",
       "      <td>3314.55</td>\n",
       "      <td>357.20</td>\n",
       "      <td>284.21</td>\n",
       "      <td>283.85</td>\n",
       "      <td>281.95</td>\n",
       "      <td>285.96</td>\n",
       "      <td>1.50</td>\n",
       "      <td>90.0</td>\n",
       "      <td>Mist</td>\n",
       "      <td>mist</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>2019-11-12</td>\n",
       "      <td>Mardi</td>\n",
       "      <td>midi</td>\n",
       "      <td>3073.50</td>\n",
       "      <td>2760.77</td>\n",
       "      <td>312.73</td>\n",
       "      <td>281.18</td>\n",
       "      <td>281.18</td>\n",
       "      <td>279.17</td>\n",
       "      <td>282.92</td>\n",
       "      <td>0.45</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Rain</td>\n",
       "      <td>light rain</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>182</th>\n",
       "      <td>2019-11-26</td>\n",
       "      <td>Mardi</td>\n",
       "      <td>midi</td>\n",
       "      <td>6.00</td>\n",
       "      <td>5.45</td>\n",
       "      <td>0.55</td>\n",
       "      <td>284.94</td>\n",
       "      <td>284.58</td>\n",
       "      <td>283.81</td>\n",
       "      <td>286.13</td>\n",
       "      <td>0.89</td>\n",
       "      <td>90.0</td>\n",
       "      <td>Rain</td>\n",
       "      <td>light rain</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>2019-12-02</td>\n",
       "      <td>Lundi</td>\n",
       "      <td>midi</td>\n",
       "      <td>2383.00</td>\n",
       "      <td>2147.05</td>\n",
       "      <td>235.95</td>\n",
       "      <td>278.87</td>\n",
       "      <td>277.13</td>\n",
       "      <td>277.49</td>\n",
       "      <td>279.87</td>\n",
       "      <td>2.24</td>\n",
       "      <td>90.0</td>\n",
       "      <td>Clouds</td>\n",
       "      <td>overcast clouds</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>2020-02-12</td>\n",
       "      <td>Mercredi</td>\n",
       "      <td>midi</td>\n",
       "      <td>3135.00</td>\n",
       "      <td>2822.80</td>\n",
       "      <td>312.20</td>\n",
       "      <td>282.08</td>\n",
       "      <td>282.08</td>\n",
       "      <td>280.47</td>\n",
       "      <td>283.02</td>\n",
       "      <td>0.89</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Clouds</td>\n",
       "      <td>broken clouds</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>2020-02-18</td>\n",
       "      <td>Mardi</td>\n",
       "      <td>midi</td>\n",
       "      <td>2105.00</td>\n",
       "      <td>1889.85</td>\n",
       "      <td>215.15</td>\n",
       "      <td>283.13</td>\n",
       "      <td>281.12</td>\n",
       "      <td>282.25</td>\n",
       "      <td>284.14</td>\n",
       "      <td>4.02</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Clouds</td>\n",
       "      <td>broken clouds</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627</th>\n",
       "      <td>2020-09-12</td>\n",
       "      <td>Samedi</td>\n",
       "      <td>midi</td>\n",
       "      <td>3998.00</td>\n",
       "      <td>3601.67</td>\n",
       "      <td>396.33</td>\n",
       "      <td>293.79</td>\n",
       "      <td>293.42</td>\n",
       "      <td>292.58</td>\n",
       "      <td>295.87</td>\n",
       "      <td>1.34</td>\n",
       "      <td>20.0</td>\n",
       "      <td>Clouds</td>\n",
       "      <td>few clouds</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>707</th>\n",
       "      <td>2020-10-20</td>\n",
       "      <td>Mardi</td>\n",
       "      <td>midi</td>\n",
       "      <td>2620.50</td>\n",
       "      <td>2352.12</td>\n",
       "      <td>268.38</td>\n",
       "      <td>287.61</td>\n",
       "      <td>287.38</td>\n",
       "      <td>286.87</td>\n",
       "      <td>288.16</td>\n",
       "      <td>0.45</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Rain</td>\n",
       "      <td>light rain</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>720</th>\n",
       "      <td>2020-10-28</td>\n",
       "      <td>Mercredi</td>\n",
       "      <td>midi</td>\n",
       "      <td>3256.50</td>\n",
       "      <td>2936.59</td>\n",
       "      <td>319.91</td>\n",
       "      <td>286.61</td>\n",
       "      <td>286.10</td>\n",
       "      <td>285.80</td>\n",
       "      <td>287.47</td>\n",
       "      <td>2.68</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Clouds</td>\n",
       "      <td>broken clouds</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>2021-07-04</td>\n",
       "      <td>Dimanche</td>\n",
       "      <td>midi</td>\n",
       "      <td>4781.00</td>\n",
       "      <td>4303.71</td>\n",
       "      <td>477.29</td>\n",
       "      <td>294.40</td>\n",
       "      <td>294.46</td>\n",
       "      <td>291.95</td>\n",
       "      <td>295.96</td>\n",
       "      <td>0.89</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Rain</td>\n",
       "      <td>light intensity shower rain</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           date      jour service   CA_TTC    CA_HT     TVA    temp  \\\n",
       "34   2019-09-18  Mercredi    midi  4077.50  3663.41  414.09  291.23   \n",
       "105  2019-10-22     Mardi    midi  3671.75  3314.55  357.20  284.21   \n",
       "152  2019-11-12     Mardi    midi  3073.50  2760.77  312.73  281.18   \n",
       "182  2019-11-26     Mardi    midi     6.00     5.45    0.55  284.94   \n",
       "195  2019-12-02     Lundi    midi  2383.00  2147.05  235.95  278.87   \n",
       "344  2020-02-12  Mercredi    midi  3135.00  2822.80  312.20  282.08   \n",
       "356  2020-02-18     Mardi    midi  2105.00  1889.85  215.15  283.13   \n",
       "627  2020-09-12    Samedi    midi  3998.00  3601.67  396.33  293.79   \n",
       "707  2020-10-20     Mardi    midi  2620.50  2352.12  268.38  287.61   \n",
       "720  2020-10-28  Mercredi    midi  3256.50  2936.59  319.91  286.61   \n",
       "811  2021-07-04  Dimanche    midi  4781.00  4303.71  477.29  294.40   \n",
       "\n",
       "     feels_like  temp_min  temp_max  wind_speed  clouds_all weather_main  \\\n",
       "34       290.22    290.36    292.47        2.24         0.0        Clear   \n",
       "105      283.85    281.95    285.96        1.50        90.0         Mist   \n",
       "152      281.18    279.17    282.92        0.45        75.0         Rain   \n",
       "182      284.58    283.81    286.13        0.89        90.0         Rain   \n",
       "195      277.13    277.49    279.87        2.24        90.0       Clouds   \n",
       "344      282.08    280.47    283.02        0.89        75.0       Clouds   \n",
       "356      281.12    282.25    284.14        4.02        75.0       Clouds   \n",
       "627      293.42    292.58    295.87        1.34        20.0       Clouds   \n",
       "707      287.38    286.87    288.16        0.45         1.0         Rain   \n",
       "720      286.10    285.80    287.47        2.68        75.0       Clouds   \n",
       "811      294.46    291.95    295.96        0.89        75.0         Rain   \n",
       "\n",
       "             weather_description  match_edf  roland_garros  fashion_week  \\\n",
       "34                  sky is clear        0.0            0.0           0.0   \n",
       "105                         mist        0.0            0.0           0.0   \n",
       "152                   light rain        0.0            0.0           0.0   \n",
       "182                   light rain        0.0            0.0           0.0   \n",
       "195              overcast clouds        0.0            0.0           0.0   \n",
       "344                broken clouds        0.0            0.0           0.0   \n",
       "356                broken clouds        0.0            0.0           0.0   \n",
       "627                   few clouds        0.0            0.0           0.0   \n",
       "707                   light rain        0.0            0.0           0.0   \n",
       "720                broken clouds        0.0            0.0           0.0   \n",
       "811  light intensity shower rain        0.0            0.0           0.0   \n",
       "\n",
       "     Match Happening  Match Happening-CL  vacances_paris  \n",
       "34               0.0                 0.0             0.0  \n",
       "105              0.0                 0.0             1.0  \n",
       "152              0.0                 0.0             0.0  \n",
       "182              0.0                 0.0             0.0  \n",
       "195              0.0                 0.0             0.0  \n",
       "344              0.0                 0.0             1.0  \n",
       "356              0.0                 0.0             1.0  \n",
       "627              0.0                 0.0             0.0  \n",
       "707              0.0                 0.0             1.0  \n",
       "720              0.0                 0.0             1.0  \n",
       "811              0.0                 0.0             0.0  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d16[d16['Match Happening-CL']==0.0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7909205f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
